From c5ca9e70d596ce10a7ac414e7a13d39441286c68 Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Fri, 11 Apr 2025 11:04:18 +0300
Subject: [PATCH] [JSW-31] mgmt: added files for jgs

As of now added files same as FS1 for
JGS asic build.

Change-Id: I22d2395331086cb2871eb4d1ab5aac272cb00f90
---

diff --git a/app/io/mgmt.c b/app/io/mgmt.c
index ec06d85..fde6976 100644
--- a/app/io/mgmt.c
+++ b/app/io/mgmt.c
@@ -14,16 +14,20 @@
 #include <zephyr/drivers/watchdog.h>
 #include <intr.h>
 #ifndef CONFIG_SOC_FS1_MGMT
+#ifndef CONFIG_SOC_JGS_MGMT
 #include <hl_boot_status.h>
+#endif
 #endif
 #include <boot_if.h>
 #include <zephyr/drivers/misc/common/common_utils.h>
 
 #ifndef CONFIG_SOC_FS1_MGMT
+#ifndef CONFIG_SOC_JGS_MGMT
 #include <hl_sensors.h>
 #include <lkd_queues.h>
 #include <hl_serdes.h>
 #include <hl_qsfp_led.h>
+#endif
 #endif
 
 #ifdef CONFIG_LIBPLDM
@@ -62,7 +66,7 @@
 EMPTY_WEAK_FUNC(void hl_cpld_i2c_recovery_init(void),)
 EMPTY_WEAK_FUNC(void hl_update_composite_temp_sensors(struct hl_dev_config *dev_config),)
 
-#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
 static void print_header(void)
 {
 	LOG_INF("*************************************");
@@ -221,7 +225,7 @@
 	return &dev_config;
 }
 
-#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
 
 #ifdef CONFIG_SHELL
 #include <zephyr/shell/shell_uart.h>
@@ -255,7 +259,7 @@
 EMPTY_WEAK_FUNC(void hl_pcie_init_drain_timeout(struct hl_dev_config *dev_config),)
 EMPTY_WEAK_FUNC(void hl_pcie_init_hbm_e2e(struct hl_dev_config *dev_config),)
 
-#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
 static int hl_device_hw_early_init(struct hl_dev_config *dev_config)
 {
 	int ret = 0;
@@ -280,7 +284,7 @@
 }
 #endif /* CONFIG_SOC_FS1_MGMT */
 
-#ifdef CONFIG_SOC_FS1_MGMT
+#if defined(CONFIG_SOC_FS1_MGMT) || defined(CONFIG_SOC_JGS_MGMT)
 __weak void sysmng_notify_boot_done(void)
 {
 	cpu_boot_sts_set(CPU_BOOT_STATUS_RUNTIME_FW_RDY);
@@ -319,7 +323,9 @@
 {
 	int32_t ret;
 
-#ifndef CONFIG_SOC_FS1_MGMT
+//#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
+
 	struct boot_status *boot_status;
 #endif
 
@@ -334,7 +340,7 @@
 		goto err_loop;
 	}
 
-#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
 	/* Set and enable HBM RLs */
 	if (is_bis_e_compliant_device())
 		hl_rtr_ctrl_init_rl_reg_once();
@@ -348,7 +354,9 @@
 		}
 	}
 
-#ifndef CONFIG_SOC_FS1_MGMT
+//#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
+
 	if (dev_config.mbx_init) {
 		ret = dev_config.mbx_init(&dev_config);
 		if (ret != 0) {
diff --git a/arch/fs1/io/device.c b/arch/fs1/io/device.c
index 9f3f9e4..25fa167 100644
--- a/arch/fs1/io/device.c
+++ b/arch/fs1/io/device.c
@@ -65,7 +65,9 @@
 }
 #endif
 
-#ifndef CONFIG_SOC_FS1_MGMT
+//#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
+
 static int fs1_mbx_init(struct hl_dev_config *config)
 {
 	return 0;
@@ -88,7 +90,9 @@
 	return 0;
 }
 
-#ifndef CONFIG_SOC_FS1_MGMT
+//#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
+
 /* To turn on Green LED(GPIO55) */
 static int fs1_status_led_on(struct hl_dev_config *dev_config)
 {
@@ -110,7 +114,9 @@
 	return 0;
 }
 
-#ifndef CONFIG_SOC_FS1_MGMT
+//#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
+
 static void fs1_init_rma_tables(struct hl_dev_config *dev_config)
 {
 }
diff --git a/arch/jgs/compute/device.c b/arch/jgs/compute/device.c
new file mode 100644
index 0000000..24d9eda
--- /dev/null
+++ b/arch/jgs/compute/device.c
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+#include <zephyr/kernel.h>
+#include <common.h>
+#include <bridge_if.h>
+#include <ipm.h>
+#include <hl_devs.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(SYSMNGD_LOG_ID);
+
+#define BRIDGE_SIZE_MIN		4
+#define BRIDGE_SIZE_MAX		36
+
+static void i2a_bridge_process_cst_wake(struct mgmt_compute_config *config)
+{
+	/* clear interrupt */
+	ipm_clear(config->bridge_wq_data.dev, 0);
+
+	/* Step1: Workpoint */
+
+	/* Step2: GO DEC */
+
+	/* Step3: GT performs Clear core state */
+
+	/*
+	 * Step4: ACK GO DEC from GT
+	 * TODO: Is it interrupt to FW? Does it indicates step3 is done?
+	 */
+
+	/* Step5: GO RESUME */
+
+	/* Step6: ACK GO DEC from GT TODO: is it interrupt to FW? */
+
+}
+
+static void i2a_bridge_process_ip_ready(struct mgmt_compute_config *config)
+{
+
+	/* clear interrupt */
+	ipm_clear(config->bridge_wq_data.dev, 0);
+
+	/* TODO: Details on TSC download command */
+}
+
+static void i2a_bridge_process_fuse_pull(struct mgmt_compute_config *config)
+{
+	/* TODO: fuse read and delivery */
+
+	/* clear interrupt */
+	ipm_clear(config->bridge_wq_data.dev, 0);
+}
+
+static void i2a_bridge_work_handler(struct k_work *work)
+{
+	struct mgmt_compute_config *config = mgmt_get_compute_cfg();
+	struct bridge_simple_msg *msg;
+	uint32_t size = config->bridge_wq_data.size;
+
+	if ((size < BRIDGE_SIZE_MIN) || (size > BRIDGE_SIZE_MAX)) {
+		LOG_ERR("Invalid size received from driver, clear interrupt");
+		ipm_clear(config->bridge_wq_data.dev, 0);
+		return;
+	}
+
+	/* decode header */
+	msg = (struct bridge_simple_msg *) config->bridge_wq_data.ptr;
+	LOG_INF("Received OPCODE: 0x%x from PORT_ID: 0x%x size: %d from I2A bridge",
+			msg->opcode, msg->source_port_id, config->bridge_wq_data.size);
+
+	/*
+	 * TODO: add processing of opcodes
+	 * some opcodes will need CR write as action to request and some
+	 * opcodes will need new response via I2C bridge in which case
+	 * create the header + data required and perform ipm_send.
+	 * CR details yet to be updated by ARCH team.
+	 */
+	switch (msg->opcode) {
+	case BRIDGE_OPCODE_CST_WAKE:
+		i2a_bridge_process_cst_wake(config);
+		break;
+	case BRIDGE_OPCODE_IP_READY:
+		i2a_bridge_process_ip_ready(config);
+		break;
+	case BRIDGE_OPCODE_FUSE_PULL:
+		i2a_bridge_process_fuse_pull(config);
+		break;
+	default:
+		LOG_ERR("OPCODE: 0x%x not supported", msg->opcode);
+		break;
+	}
+
+	/* clear interrupt */
+	ipm_clear(config->bridge_wq_data.dev, 0);
+}
+
+static void i2a_bridge_callback(const struct device *dev,
+				void *context, uint32_t size, void *data)
+{
+	struct mgmt_compute_config *config = mgmt_get_compute_cfg();
+
+	config->bridge_wq_data.size = size;
+	config->bridge_wq_data.ptr = (void *) data;
+	config->bridge_wq_data.dev = dev;
+	k_work_submit(&config->bridge_work);
+}
+
+static int fs_init_bridge(void)
+{
+	struct mgmt_compute_config *config = mgmt_get_compute_cfg();
+
+	config->bridge_dev = I2A_DEV_INST;
+	if (!config->bridge_dev) {
+		LOG_ERR("Failed to get pointer to I2A bridge device!");
+		return -ENODEV;
+	}
+
+	k_work_init(&config->bridge_work, i2a_bridge_work_handler);
+
+	ipm_register_callback(config->bridge_dev, i2a_bridge_callback, config);
+	ipm_set_enabled(config->bridge_dev, true);
+	return 0;
+}
+
+
+int mgmt_dev_prepare(void)
+{
+	struct mgmt_compute_config *config = mgmt_get_compute_cfg();
+
+	config->init_bridge = fs_init_bridge;
+	return 0;
+}
diff --git a/arch/jgs/dev_queue_packets.c b/arch/jgs/dev_queue_packets.c
new file mode 100644
index 0000000..177600a
--- /dev/null
+++ b/arch/jgs/dev_queue_packets.c
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+#include <dev_queue_packets.h>
+#include <cpucp_resp_if.h>
+
+#define LOG_LEVEL                       CONFIG_LOG_DEFAULT_LEVEL
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(fs1_queues_packet);
+
+/* TODO: Remove this once frequency is retrieved from pll driver. */
+struct fs1_pll_info pll_info_array[CPUCP_PLL_IDX_MAX] = {
+	{CPUCP_PLL_PCIE, 2000, 1000, 500, 100},
+	{CPUCP_PLL_PSOC, 1000, 500, 200, 50},
+	{CPUCP_PLL_NIC, 800, 533, 400, 100},
+	{CPUCP_PLL_DEC, 1000, 500, 500, 125},
+	{CPUCP_PLL_ENC, 1000, 500, 500, 125},
+	{CPUCP_PLL_HBM, 1600, 800, 200, 100},
+	{CPUCP_PLL_GT_CU, 1600, 1600, 400, 200},
+	{CPUCP_PLL_GT_SB, 1200, 600, 400, 25},
+	{CPUCP_PLL_DMA, 1600, 800, 400, 200},
+	{CPUCP_PLL_C2C, 2000, 1000, 500, 200},
+	{CPUCP_PLL_D2D, 2000, 667, 500, 200},
+	{CPUCP_PLL_SGAM, 1600, 1600, 1600, 1600}
+};
+
+int queue_dev_get_pll_index_info(uint32_t pll_index, struct cpucp_resp_pll_info_get *pll_info_get)
+{
+
+	if (pll_index == 0 || pll_index > CPUCP_PLL_IDX_MAX)
+		return -EINVAL;
+
+	pll_info_get->size = sizeof(struct cpucp_resp_pll_info_get);
+	pll_info_get->div0_freq = pll_info_array[pll_index - 1].div0_freq;
+	pll_info_get->div1_freq = pll_info_array[pll_index - 1].div1_freq;
+	pll_info_get->div2_freq = pll_info_array[pll_index - 1].div2_freq;
+	pll_info_get->div3_freq = pll_info_array[pll_index - 1].div3_freq;
+
+	return 0;
+}
diff --git a/arch/jgs/io/device.c b/arch/jgs/io/device.c
new file mode 100644
index 0000000..f480921
--- /dev/null
+++ b/arch/jgs/io/device.c
@@ -0,0 +1,248 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+/* IMPORTANT:
+ * This file is needed for basic arcmgmt implementation.
+ * The filename and its contents was copied from the
+ * corresponding file under gaudi3 directory.
+ * Throughout the process of the arcmgmt implementation,
+ * it will be validated (i.e. discarded or modified).
+ * Refer to ticket FSW-6865.
+ *
+ */
+
+#include <common.h>
+#include <zephyr/drivers/gpio.h>
+#include <rma_if.h>
+#include <zephyr/drivers/i2c.h>
+#include <common_utils.h>
+#include <app_defs.h>
+#include <boot_if.h>
+#include <global_conf_regs.h>
+#include <jgs_reg_map.h>
+#include <arc_regs.h>
+#include <zephyr/drivers/misc/hl_cpld/hl_cpld.h>
+
+#include <arc_regs.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(SYSMNGD_LOG_ID);
+
+static struct mgmt_dev_data dev_data;
+
+#define PCI_IDS_FS1				0x0b73
+#define PCI_IDS_FS1_SIMULATOR			0x0b7a
+#define PCI_IDS_FS1_SIMULATOR_SINGLE_DIE	0x0b7b
+#define PCI_IDS_FS1_FPGA			0x0b7c
+
+#define DEVICE_POLLING_DELAY		1 /* mSec */
+
+#define NIC_REQ_NIC_STS_PERIODIC_SEC	5
+#define NIC_REQUEST_START_CMD		1
+
+#define STATUS_LED_GPIO_PIN		7
+#define GPIO_PIN_LEVEL_HIGH		1
+
+/*
+ * TODO: the below section will need to be reworked for FS1
+ * Currently it is not called from FS1 code.
+ */
+#ifdef TO_BE_REMOVED
+enum hl_pci_ids pcie_dev_ids[PCIE_DEV_ID_SIZE] = { PCI_IDS_FS1,
+						   PCI_IDS_FS1_SIMULATOR,
+						   PCI_IDS_FS1_SIMULATOR_SINGLE_DIE,
+						   PCI_IDS_FS1_FPGA,
+						};
+
+enum hl_pci_ids hl_dev_get_fpga_pcie_dev_id(void)
+{
+	return PCI_IDS_FS1;
+}
+#endif
+
+//#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
+
+static int fs1_mbx_init(struct hl_dev_config *config)
+{
+	return 0;
+}
+#endif
+
+int hl_device_hw_early_init(struct hl_dev_config *dev_config)
+{
+	return 0;
+}
+
+int fs1_security_init(struct hl_dev_config *dev_config)
+{
+	LOG_INF("Security currently bypassed in FS1");
+	return 0;
+}
+
+static int fs1_wait_for_arcs(struct hl_dev_config *dev_config)
+{
+	return 0;
+}
+
+//#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
+
+/* To turn on Green LED(GPIO55) */
+static int fs1_status_led_on(struct hl_dev_config *dev_config)
+{
+	return gpio_pin_set(dev_config->led_gpio, STATUS_LED_GPIO_PIN, GPIO_PIN_LEVEL_HIGH);
+}
+#endif
+
+int hl_device_hw_init(struct hl_dev_config *dev_config)
+{
+	return 0;
+}
+
+void hl_sensors_data_init(void)
+{
+}
+
+int hl_init_hbm_mem_pool(struct hl_dev_config *dev_config)
+{
+	return 0;
+}
+
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
+static void fs1_init_rma_tables(struct hl_dev_config *dev_config)
+{
+}
+#endif
+
+void nic_status_com_init(struct hl_dev_config *dev_config)
+{
+}
+
+static bool i2c_recovery_in_process;
+
+__weak int hl_send_i2c_filter_recovery_request(const struct device *gpio_dev, uint8_t gpio_pin)
+{
+	return 0;
+}
+
+static bool is_i2c_recovery_available(const struct device *i2c_dev)
+{
+	const struct cpld_i2c_recovery_api *i2c_recovery_api;
+	bool res = false;
+
+	i2c_recovery_api = get_cpld_i2c_recovery_api();
+
+	if (i2c_recovery_api)
+		res = i2c_recovery_api->is_cpld_recovery_available(i2c_dev);
+
+	return res;
+}
+
+static int cpld_i2c_recovery_event_handler(const struct device *i2c_dev)
+{
+	if (i2c_recovery_in_process)
+		return 0;
+
+	if (!is_i2c_recovery_available(i2c_dev))
+		return 1;
+
+	i2c_recovery_in_process = true;
+
+	i2c_recovery_in_process = false;
+
+	return 0;
+}
+
+void hl_cpld_i2c_recovery_init(void)
+{
+	const struct cpld_i2c_recovery_api *i2c_recovery_api;
+
+	i2c_recovery_api = get_cpld_i2c_recovery_api();
+
+	if (i2c_recovery_api)
+		i2c_recovery_api->set_recovery_event_handler(cpld_i2c_recovery_event_handler);
+	else
+		LOG_ERR("Failed to get I2C recovery API");
+}
+
+__weak bool fs1_run_engine_arcs_init(struct hl_dev_config *dev_config)
+{
+	return true;
+}
+
+void cpu_boot_if_dev_spec_regs_init(struct cpu_boot_if_regs *regs)
+{
+	/* boot_if module wants the full address */
+	regs->status = HL_CFG_BASE + IO0_H0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_CPU_BOOT_STATUS);
+	regs->response = HL_CFG_BASE + mmCPU_CMD_STATUS_TO_HOST;
+	regs->kmd_msg_to_cpu = HL_CFG_BASE + IO0_H0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_KMD_MSG_TO_CPU);
+	regs->hw_state = HL_CFG_BASE + mmHW_STATE;
+	regs->err[0] = HL_CFG_BASE + mmCPU_BOOT_ERR0;
+	regs->err[1] = HL_CFG_BASE + mmCPU_BOOT_ERR1;
+	regs->dev_sts[0] = HL_CFG_BASE + mmCPU_BOOT_DEV_STS0;
+	regs->dev_sts[1] = HL_CFG_BASE + mmCPU_BOOT_DEV_STS1;
+}
+
+int mgmt_dev_prepare(void)
+{
+	struct hl_dev_config *dev_config = hl_get_dev_cfg();
+
+	LOG_INF("fs1 device prepare called");
+
+	dev_config->dev_data = &dev_data;
+	dev_config->queue_init = fs_init_queues;
+	dev_config->wait_for_arcs = fs1_wait_for_arcs;
+
+	return 0;
+}
+
+void hl_pcie_init_drain_timeout(struct hl_dev_config *dev_config) {}
+void hl_pcie_init_hbm_e2e(struct hl_dev_config *dev_config) {}
+void hl_dev_spec_dis_pci_access(void) {}
+
+int hl_unmask_nic_sts_events(uint32_t *event_ids, unsigned int len)
+{
+	return 1;
+}
+
+void preboot_reset_init(void)
+{
+}
+
+void init_engines_ctx(void) {}
+void hl_hbm_enable_dbg_counters(struct hl_dev_config *dev_config) {}
+void ras_init(struct hl_dev_config *dev_config) {}
+void hl_dev_spec_en_pci_access(void) {}
+
+#define FS_QUEUE_SCRATCHPAD_ADDR_HI	0xF8830000
+#define FS_QUEUE_SCRATCHPAD_ADDR_LO	0xF8820000
+
+/* TODO: change these macros according to specs file */
+#define FS_QUEUE_SPSRAM_EXTEN_ADDR	0x70000000
+#define FS_QUEUE_HBW_EXTN_ADDR		0x10000000
+void queue_dev_update_arc_addr_extn(uint32_t addr_hi, uint32_t addr_lo,
+				    uint32_t *extn_addr, bool *extn_set)
+{
+	struct hl_dev_config *dev_config = hl_get_dev_cfg();
+
+	if (addr_lo >= FS_QUEUE_SCRATCHPAD_ADDR_LO && addr_hi <= FS_QUEUE_SCRATCHPAD_ADDR_HI)
+		*extn_addr = FS_QUEUE_SPSRAM_EXTEN_ADDR;
+	else {
+		CFG_WREG32((mmIO_0_CFG_PSOC_ARC_MNG_BASE + mmARC_HBW_OFFSET_REGION_1), 0x0);
+		CFG_WREG32((mmIO_0_CFG_PSOC_ARC_MNG_BASE + mmARC_HBW_MSB_ADDR_REGION_1),
+			addr_hi);
+		CFG_WREG32((mmIO_0_CFG_PSOC_ARC_MNG_BASE + mmARC_HBW_LSB_ADDR_REGION_1),
+			addr_lo & 0xFFFFFFF);
+
+		*extn_addr = FS_QUEUE_HBW_EXTN_ADDR;
+	}
+
+	*extn_set = true;
+	dev_config->boot_done = true;
+}
diff --git a/include/arch/jgs/app_defs.h b/include/arch/jgs/app_defs.h
new file mode 100644
index 0000000..20f28a5
--- /dev/null
+++ b/include/arch/jgs/app_defs.h
@@ -0,0 +1,27 @@
+/* SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+/* IMPORTANT:
+ * This file is needed for basic arcmgmt implementation.
+ * The filename and its contents was copied from the
+ * corresponding file under gaudi3 directory.
+ * Throughout the process of the arcmgmt implementation,
+ * it will be validated (i.e. discarded or modified).
+ * Refer to ticket FSW-6865.
+ *
+ */
+
+#ifndef ZEPHYR_INCLUDE_FS1_DEFS_H
+#define ZEPHYR_INCLUDE_FS1_DEFS_H
+
+#include <queue_comms.h>
+
+struct mgmt_dev_data {
+	/* TODO: add fs1 device specific variables & data structure(s) */
+	struct fs_queues_config queues_cfg;
+};
+
+#endif /* ZEPHYR_INCLUDE_FS1_DEFS_H */
diff --git a/include/arch/jgs/dev_queue_packets.h b/include/arch/jgs/dev_queue_packets.h
new file mode 100644
index 0000000..ea66e82
--- /dev/null
+++ b/include/arch/jgs/dev_queue_packets.h
@@ -0,0 +1,23 @@
+/* SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+#ifndef ZEPHYR_INCLUDE_FS1_QUEUE_PACKETS_H
+#define ZEPHYR_INCLUDE_FS1_QUEUE_PACKETS_H
+
+#include <cpucp_resp_if.h>
+#include <cpucp_common_if.h>
+
+struct fs1_pll_info {
+	enum cpucp_pll_index pll_index;
+	uint16_t div0_freq;
+	uint16_t div1_freq;
+	uint16_t div2_freq;
+	uint16_t div3_freq;
+};
+
+int queue_dev_get_pll_index_info(uint32_t pll_index, struct cpucp_resp_pll_info_get *pll_info_get);
+
+#endif /* ZEPHYR_INCLUDE_FS1_QUEUE_PACKETS_H */
diff --git a/include/arch/jgs/polling.h b/include/arch/jgs/polling.h
new file mode 100644
index 0000000..eedbb74
--- /dev/null
+++ b/include/arch/jgs/polling.h
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+/* IMPORTANT:
+ * This file is needed for basic arcmgmt implementation.
+ * The filename and its contents was copied from the
+ * corresponding file under gaudi3 directory.
+ * Throughout the process of the arcmgmt implementation,
+ * it will be validated (i.e. discarded or modified).
+ * Refer to ticket FSW-6865.
+ *
+ */
+
+#ifndef ZEPHYR_INCLUDE_FS1_POLLING_H
+#define ZEPHYR_INCLUDE_FS1_POLLING_H
+
+#endif /* ZEPHYR_INCLUDE_FS1_POLLING_H */
diff --git a/include/io/common.h b/include/io/common.h
index 24ac28e..3083866 100644
--- a/include/io/common.h
+++ b/include/io/common.h
@@ -20,7 +20,12 @@
 #include "eeprom_util.h"
 #include <zephyr/drivers/sensor.h>
 
-#ifdef CONFIG_SOC_FS1_MGMT
+//#ifdef CONFIG_SOC_FS1_MGMT
+#if defined(CONFIG_SOC_FS1_MGMT) || defined(CONFIG_SOC_JGS_MGMT)
+
+#include <cpucp_common_if.h>
+#include <hl_boot_if.h>
+#elif CONFIG_SOC_JGS_MGMT
 #include <cpucp_common_if.h>
 #include <hl_boot_if.h>
 #else
@@ -138,7 +143,9 @@
 	struct pcie_aux_register_offset pcie_aux_reg_off;
 	int ctemp_fixup;
 	bool legacy_boot;
-#ifndef CONFIG_SOC_FS1_MGMT
+//#ifndef CONFIG_SOC_FS1_MGMT
+#if !defined(CONFIG_SOC_FS1_MGMT) && !defined(CONFIG_SOC_JGS_MGMT)
+
 	struct cpld_version_data cpld_ver;
 #endif
 	struct isolation_cfg iso_cfg;
@@ -282,7 +289,7 @@
 void hl_pcie_init_drain_timeout(struct hl_dev_config *dev_config);
 void hl_pcie_init_hbm_e2e(struct hl_dev_config *dev_config);
 void hl_rtr_ctrl_init_rl_reg_once(void);
-#ifdef CONFIG_SOC_FS1_MGMT
+#if defined(CONFIG_SOC_FS1_MGMT) || defined(CONFIG_SOC_JGS_MGMT)
 int fs_init_queues(struct hl_dev_config *dev_config);
-#endif /* CONFIG_SOC_FS1_MGMT */
+#endif /* CONFIG_SOC_FS1_MGMT  CONFIG_SOC_JGS_MGMT */
 #endif /* ZEPHYR_INCLUDE_SYSMNGD_COMMON_H_ */
